src/frontends/tensorflow/src/op/space_to_batch_nd.cpp

```diff
@@ -36,7 +36,13 @@ OutputVector translate_batch_nd_and_space_nd_op(const NodeContext& node) {
     }
 
     auto N = input_pshape.rank().get_length();
-    auto M = block_shape_pshape.rank().get_length();
+
+    // TODO: support dynamic shape
+    TENSORFLOW_OP_VALIDATION(node,
+                             block_shape_pshape.is_static(),
+                             "Dynamic shape for block_shape input is not supported.");
+    auto M = static_cast<int64_t>(block_shape_pshape.to_shape()[0]);
```

Comment:
If you need only 0 dimension you can get it from the partial shape and then check that dimension is static. That way you restrict one dimension instead of all shape

Reply:
Yes, I know, but this is 1D input, only one dimension is allowed according to tf spec

Reply:
Hmm, ok. I still think that as I suggested it will be cleaner, but it is nice to have, not a blocker.

Reply:
fixed